packages/ra-core/src/form/FormWithRedirect.spec.tsx

```diff
@@ -41,7 +41,7 @@ describe('FormWithRedirect', () => {
 
         expect(renderProp.mock.calls[1][0].pristine).toEqual(true);
         expect(getByDisplayValue('Foo')).not.toBeNull();
-        expect(renderProp).toHaveBeenCalledTimes(2);
+        expect(renderProp).toHaveBeenCalledTimes(3);
```

Comment:
Using `useFormState` triggers a rerender.

Reply:
That's a problem... This feature impacts the performance of all forms to allow a feature that isn't so common...